### ISSUE TITLE
Composer install problem

### ISSUE THREAD

**Issue by svmt** [mollie/PrestaShop#78]
Composer require/update/install problem

Composer is unable to pull repository. It's looping in attempt to download version downgrading version on each unsuccessful request till you reach github rate limit 

```
$ composer install
Loading composer repositories with package information                           
Reading composer.json of firstred/mollie-api-php (v2.1.5)
Reading composer.json of firstred/mollie-api-php (v2.1.4)
Reading composer.json of firstred/mollie-api-php (v2.1.3)
...
Reading composer.json of firstred/mollie-api-php (v1.9.5)GitHub API limit (60 calls/hr) is exhausted, could not fetch https://api.github.com/repos/firstred/mollie-api-php/commits/75669c82aac7601b7fef1fe103533a3afe041782
```

### Environment
* PrestaShop version: 1.6.17
* Mollie module version: 3.3.2
* PHP version 7.1
* Composer version 1.7.3

**Comment by firstred** [comment]
Hi @svmt 

Thanks for reaching out!

GitHub by default limits its API to 60 calls per hour and once you hit this limit you can no longer download packages from GitHub directly. The PHP library (firstred/mollie-api-php) is a fork with lower requirements (PHP5.3+) and is not published on [packagist](https://packagist.org/).

In order to raise the GitHub limit you will have to authenticate first. Here's a link that should help:  https://getcomposer.org/doc/articles/troubleshooting.md#api-rate-limit-and-oauth-tokens

Note that you can also authenticate with your SSH key, for this you should modify the `repositories` section in the `composer.json` file:
```
 "repositories": [
        {
            "type": "vcs",
            "url": "git@github.com:firstred/mollie-api-php"
        },
        {
            "type": "vcs",
            "url": "git@github.com:firstred/mollie-reseller-api"
        }
    ],
```
Are you looking to make your own customized build? There are a few more steps to complete before you have the entire build. You can check out the [Makefile](https://github.com/mollie/PrestaShop/blob/3155c62a007c80c625eeabf5e91e77efc2c2b136/Makefile) to see all the steps. `make dev` will build a development version whereas `make prod` will build a full release suitable for production.

**Comment by svmt** [comment]
Hi @firstred 
no i was trying to update to the latest version 3.3.2 
Official zip Release 3.3.1 works fine to install but has a problem so i was just trying to test latest release to see if it was addressed

```
 An error occurred:
    Error executing API call (422: Unprocessable Entity): Order line 1 is invalid. Quantity must be greater than 0.. Field: lines.1.quantity. Documentation: https://docs.mollie.com/guides/handling-errors
```
this error is raised due to QTY = 0 and it is set in mollie::getCartLines()
if you have 1 item you have a problem since on line 2256: 'quantity'    => $quantity - 1,

```
    if ($lastItemPriceDifference >= 0.01 || $lastItemTaxDifference >= 0.01) {
                $aItems[] = array(
                    'name'        => $cartItem['name'],
                    'quantity'    => $quantity - 1,
```

1-1 = 0

i guess i would need to create a new issue?

**Comment by firstred** [comment]
That is interesting. It it trying to compensate for the rounding method used and making it Klarna-compatible, but I wasn't aware it could lead to `qty=0`. You can try the beta in this comment: https://github.com/mollie/PrestaShop/issues/73#issuecomment-435534685, which is the latest version since we haven't pushed any commits since, but my guess is that the problem is still there.

We'll get started on this issue right away!

**Comment by svmt** [comment]
OK thanks for acknowledging the issue even so it turned out unrelated to the original topic 
Anyways i'll wait for it to be fixed

**Comment by firstred** [comment]
Thanks for the information!

It looks like it is trying to compensate for the tax difference caused by rounding whereas the module should accept this difference on every line. My guess is that an additional quantity check should be sufficient: ` if (($lastItemPriceDifference >= 0.01 || $lastItemTaxDifference >= 0.01) && $quantity > 1) {`, but we'll need to have a further look at that.

Meanwhile here's a new build you can try: 
[mollie-v3.3.2_20181105153724.zip](https://github.com/mollie/PrestaShop/files/2548829/mollie-v3.3.2_20181105153724.zip)

**Comment by svmt** [comment]
this doesn't really solve the issue by adding  && $quantity > 1
Since the amount is not what expected another error is raised:

Error executing API call (422: Unprocessable Entity): The amount of the order does not match the total amount from the order lines. Expected order amount to be €40.48 but got €40.47. Field: amount. Documentation: https://docs.mollie.com/guides/handling-errors

also upgrading from zip: mollie-v3.3.2_20181105153724.zip
seems function name is wrong should be 3_3_2 instead of 3_3_0 isn't it?
here is a screenshot :
![2018-11-06 1](https://user-images.githubusercontent.com/19215202/48060083-3474bd80-e1bb-11e8-8605-6bf72a9e9393.png)

**Comment by svmt** [comment]
@firstred i'll be waiting for the solution to rounding problem since we require OrderAPI to use Klarna payments, could you let me know when to expect your solution so i can plan if i should park the issue for the time?

**Comment by firstred** [comment]
Marking this as solved since v3.3.3. Feel free to reopen if you still have this problem with the latest version (v3.3.5 at the moment).
